okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpEngine.java

```diff
@@ -899,7 +899,15 @@ public void readResponse() throws IOException {
         bufferedRequestBody.close();
       }
 
-      return readNetworkResponse();
+      Response response = readNetworkResponse();
+
+      int code = response.code();
+      if ((code == 204 || code == 205) && response.body().contentLength() > 0) {
```

Comment:
Yeah, this doesn't feel like the right place for it. This has nothing to do with interceptors! Perhaps better in readNetworkResponse(), guarded by `callerWritesRequestBody` ? (That's code for `HttpUrlConnection`.)